DANS-KNAW/easy-deposit-api
# EASY-1464 submit

Fixes EASY-1464 submit

#### When applied it will
* [x] integration test nor the manual test should use the `draft` folder for `easy-ingest-flow-inbox`
  On the right hand side below `drafts` should have been `easy-ingest-flow-inbox`  
  https://github.com/DANS-KNAW/easy-deposit-api/blob/1f180c26480cf05ebfa0fd61a402c2b223079dc6/src/test/scala/nl.knaw.dans.easy.deposit/servlets/IntegrationSpec.scala#L207-L209

#### Where should the reviewer @DANS-KNAW/easy start?

#### How should this be manually tested?

A submit should now result in success once the metatadata is valid.

From #38 and #47 : use deasy because the local VM has no pid-generator:

1) build, deploy an create a draft deposit:
```
cd easy-deposit-api
mvn clean install
cd easy-dtap
vagrant destroy -f && vagrant up --no-provision ### only for a fresh base-box
./deploy-role.sh easy-deposit-api
curl -i -u user001:user001 -X POST http://deasy:20190/deposit
```

2) Copy the UUID returned by the curl command above. 
3) No curl command figured out to put json content, use PostMan to add metadata and submit:
    * put metadata from one of the files in `easy-deposit-api/src/test/resources/manual-test`
      but remove the DOI from the identifiers
      to `http://localhost:20190/deposit/<UUID>/metadata`
    * put `{"state":"SUBMITTED","stateDescription":"blabla"}`
      to `http://localhost:20190/deposit/<UUID>/state`

The results in `/var/opt/dans.knaw.nl/` on deasy:
* `deposit/drafts/user001/<UUID>` should contain the draft
* `deposit/stage/<UUID>` should contain a copy while processing the state change
* `tmp/easy-ingest-flow-inbox/<UUID>` should contain a copy of the draft with different files in `metadata`

#### Related pull requests on github
* [x] https://github.com/DANS-KNAW/dans-bag-lib/milestone/1
